### PR TITLE
Use pre-built python3 binary on OSX

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
I think I finally figured this out!

File: scripts/update_python.py [hunk 1]
```diff
@@ -61,23 +67,56 @@ def make_python_patch(arch):
...
-def main():
-    for arch in ('amd64', 'win32'):
-        make_python_patch(arch)
+def build_python():
+    if sys.platform.startswith('darwin'):
+        osname = 'macos'
+        # Take some rather drastic steps to link openssl statically
+        check_call(['brew', 'install', 'openssl', 'pkg-config'])
+        os.remove('/usr/local/opt/openssl/lib/libssl.dylib')
+        os.remove('/usr/local/opt/openssl/lib/libcrypto.dylib')
```
Comment (kripken) [inline scripts/update_python.py:89]: do these require root?

Comment (sbc100) [reply scripts/update_python.py:89]: No, apparently on macOS normaly users have control of /usr/local.

See https://docs.brew.sh/FAQ#why-does-homebrew-prefer-i-install-to-usrlocal

Comment (kripken) [reply scripts/update_python.py:89]: Interesting, I did not know that.

**Pull request merged**